### PR TITLE
Limit orders/receipt other fixes

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

This PR fixes few issues mentioned in other PR related to receipt
- removes moment.js library and uses date-fns
- fixes the issues mentioned by @elena-zh  here https://github.com/cowprotocol/cowswap/pull/1502#issuecomment-1330735315 and https://github.com/cowprotocol/cowswap/pull/1502#issuecomment-1330524745 for point 1
- also adds some code improvements mentioned in PR's

### CODE REVIEW

**Comment by alfetopito** [comment]
> # Summary
> 
> This PR fixes few issues mentioned in other PR related to receipt
> 
>     * removes moment.js library and uses date-fns
> 
>     * fixes the issues mentioned by @elena-zh  here [Limit orders/receipt #1502 (comment)](https://github.com/cowprotocol/cowswap/pull/1502#issuecomment-1330735315) and [Limit orders/receipt #1502 (comment)](https://github.com/cowprotocol/cowswap/pull/1502#issuecomment-1330524745) for point 1
> 
>     * also adds some code improvements mentioned in PR's

Let's try to keep the scope small
Each item could have a separated and smaller PR by itself, no? ;)

File: src/cow-react/modules/limitOrders/pure/ReceiptModal/DateField.tsx [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+import { format } from 'date-fns'
 
 interface Props {
-  date: string | Date | undefined
+  date: Date | undefined
 }
 
 export function DateField({ date }: Props) {
+  const timeAgo = useTimeAgo(date)
+  const previewDate = format(date || 0, 'MMM d YYY, h:mm a')
```
Comment (alfetopito) [inline src/cow-react/modules/limitOrders/pure/ReceiptModal/DateField.tsx:11]: Would [this](https://github.com/cowprotocol/cowswap/blob/7dc13801002e667e2625f4d51d30cf294c0d483f/src/cow-react/utils/time.ts#L15) help here?

**Comment by fleupold** [comment]
Because the whole cowswap dev team was used for review (rather than just frontend)

**Comment by nenadV91** [comment]
My apologies :sweat_smile:

File: src/cow-react/modules/limitOrders/pure/ReceiptModal/DateField.tsx (hunk 1 above)
Comment (nenadV91) [reply src/cow-react/modules/limitOrders/pure/ReceiptModal/DateField.tsx:11]: I don't think so, if we want this specific format

**Review by elena-zh — APPROVED** [review]
LGTM!

File: src/cow-react/modules/limitOrders/pure/ReceiptModal/OrderTypeField.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+export type Props = {
+  order: ParsedOrder
+}
+
+export function OrderTypeField({ order }: Props) {
+  return (
+    <styledEl.Value>
+      <styledEl.OrderTypeValue>
+        {order.class} {order.kind} order {!order.partiallyFillable && '(Fill or Kill)'}
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply src/cow-react/modules/limitOrders/pure/ReceiptModal/OrderTypeField.tsx:12]: Agree, it's related to: https://github.com/cowprotocol/explorer/pull/259/files

**Review by shoom3301 — APPROVED** [review]

File: src/cow-react/modules/limitOrders/pure/ReceiptModal/DateField.tsx (hunk 1 above)
Comment (anxolin) [reply src/cow-react/modules/limitOrders/pure/ReceiptModal/DateField.tsx:11]: Why is this case we want a specific date format?

**Pull request merged**